pygeos/test/test_creation.py

```diff
@@ -318,3 +318,26 @@ def test_subclasses(with_point_in_registry):
         assert isinstance(point, Point)
         assert pygeos.get_type_id(point) == pygeos.GeometryType.POINT
         assert point.x == 1
+
+
+def test_prepare():
+    arr = np.array([pygeos.points(1, 1), None, pygeos.box(0, 0, 1, 1)])
+    pygeos.prepare(arr)
+    assert arr[0]._ptr_prepared != 0
+    assert arr[1] is None
+    assert arr[2]._ptr_prepared != 0
+
+    # preparing again actually does nothing
+    original = arr[0]._ptr_prepared
+    pygeos.prepare(arr)
+    assert arr[0]._ptr_prepared == original
+
+
+def test_destroy_prepared():
+    arr = np.array([pygeos.points(1, 1), None, pygeos.box(0, 0, 1, 1)])
+    pygeos.prepare(arr)
+    pygeos.destroy_prepared(arr)
```

Comment:
Even though a bit redundant with above test, for internal consistency here, probably a good idea to validate that it is actually prepared before `destroy_prepared`:
```
    assert arr[0]._ptr_prepared != 0
    assert arr[2]._ptr_prepared != 0
    pygeos.destroy_prepare(arr)
```